docs/api/api_blueprint.apib

```diff
@@ -4422,6 +4421,327 @@ Delete a single notification.
+    + sort (optional, string, `name`) - Sort the resources according to the given attribute in ascending order. Append '-' to sort in descending order.
+    + filter (optional, string, `[{"type": "percent" }]`) - Filter according to the flask-rest-jsonapi filtering system. Please refer: http://flask-rest-jsonapi.readthedocs.io/en/latest/filtering.html for more.
+
+### List All Discount Codes [GET]
+Get a list of Discount Codes.
+
++ Request
+
+    + Headers
+
+            Accept: application/vnd.api+json
+
+            Authorization: JWT <Auth Key>
+
++ Response 200 (application/vnd.api+json)
+
+        {
+          "meta": {
+              "count": 1
+          },
+          "data": [
+              {
+                  "relationships": {
+                      "event": {
+                          "links": {
+                              "self": "/v1/discount-codes/1/relationships/event",
+                              "related": "/v1/discount-codes/1/event"
+                          }
+                      }
+                  },
+                  "attributes": {
+                      "tickets": null,
+                      "code": "DC101",
+                      "valid-from": "2017-06-18T18:30:00+00:00",
+                      "min-quantity": 0,
+                      "created-at": "2017-06-20T09:59:23.740772+00:00",
+                      "tickets-number": 404,
+                      "value": 100,
+                      "max-quantity": 100,
+                      "is-active": false,
+                      "used-for": "Event",
+                      "valid-till": "2017-06-24T18:30:00+00:00",
+                      "type": "amount",
+                      "discount-url": "https://my-discount-url.com"
+                  },
+                  "type": "discount-code",
+                  "id": 1,
+                  "links": {
+                      "self": "/v1/discount-codes/1"
+                  }
+              }
+          ],
+          "jsonapi": {
+              "version": "1.0"
+          },
+          "links": {
+              "self": "/v1/events/1/discount-codes"
+          }
+         }
+
+### Create Discount Code [POST]
+Create a new Discount Code with event_id.
+
++ Request (application/vnd.api+json)
+
+    + Headers
+
+            Authorization: JWT <Auth Key>
+
+    + Body
+
+            {
+              "data": {
+                "attributes": {
+                  "code": "DC101",
+                  "discount-url": "https://my-discount-url.com",
+                  "value": "100",
+                  "type": "amount",
+                  "is_active": "false",
+                  "tickets_number": "404",
+                  "min_quantity": "0",
+                  "max_quantity": "100",
+                  "valid_from": "2017-06-18T18:30:00+00:00",
+                  "valid_till": "2017-06-24T18:30:00+00:00",
+                  "used_for": "Event"
```

Comment:
@enigmaeth 
These need to be dasherized.
min-quantity , max-quantity , valid-from , valid-till , used-for